crates/astria-sequencer/src/accounts/query.rs

```diff
@@ -215,3 +223,84 @@ async fn preprocess_request(
     };
     Ok((address, snapshot, height))
 }
+
+/// Custom deterministic sorting function for `AssetBalance` that sorts by balance in descending
+/// order and then by denom in ascending order. This function should never return `Ordering::Equal`,
+/// as calling `sort_unstable_by` on equal values has a non-deterministic result.
+fn compare_asset_balances(lhs: &AssetBalance, rhs: &AssetBalance) -> std::cmp::Ordering {
```

Comment:
Rethinking this: these should just be sorted by lexical order of the denominations.

You cannot compare the magnitude of different quantities, so we shouldn't order by them.